From 5b72cc0dd3ce0288408ee0a1b0785f7c603a059e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fl=C3=B6ssie?= <floessie.mail@gmail.com>
Date: Tue, 24 Sep 2019 09:59:31 +0200
Subject: [PATCH] Quote parameters correctly for Linux when spawning (#5463)

---
 rtgui/extprog.cc | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/rtgui/extprog.cc b/rtgui/extprog.cc
index a6a9050c0..95c1c937d 100644
--- a/rtgui/extprog.cc
+++ b/rtgui/extprog.cc
@@ -58,7 +58,7 @@ bool ExtProgAction::execute (const std::vector<Glib::ustring>& fileNames) const
     }
 
     for (const auto& fileName : fileNames) {
-        cmdLine += " \"" + fileName + "\"";
+        cmdLine += " " + Glib::shell_quote(fileName);
     }
 
     return ExtProgStore::spawnCommandAsync (cmdLine);
@@ -256,7 +256,7 @@ bool ExtProgStore::openInGimp (const Glib::ustring& fileName)
 
 #else
 
-    auto cmdLine = Glib::ustring("gimp \"") + fileName + Glib::ustring("\"");
+    auto cmdLine = Glib::ustring("gimp ") + Glib::shell_quote(fileName);
     auto success = spawnCommandAsync (cmdLine);
 
 #endif
@@ -291,7 +291,7 @@ bool ExtProgStore::openInGimp (const Glib::ustring& fileName)
 
 #else
 
-    cmdLine = Glib::ustring("gimp-remote \"") + fileName + Glib::ustring("\"");
+    cmdLine = Glib::ustring("gimp-remote ") + Glib::shell_quote(fileName);
     success = ExtProgStore::spawnCommandAsync (cmdLine);
 
 #endif
@@ -312,7 +312,7 @@ bool ExtProgStore::openInPhotoshop (const Glib::ustring& fileName)
 
 #else
 
-    const auto cmdLine = Glib::ustring("\"") + Glib::build_filename(options.psDir, "Photoshop.exe") + Glib::ustring("\" \"") + fileName + Glib::ustring("\"");
+    const auto cmdLine = Glib::ustring("\"") + Glib::build_filename(options.psDir, "Photoshop.exe") + "\" " + Glib::shell_quote(fileName);
 
 #endif
 
@@ -334,7 +334,7 @@ bool ExtProgStore::openInCustomEditor (const Glib::ustring& fileName)
 
 #else
 
-    const auto cmdLine = Glib::ustring("\"") + options.customEditorProg + Glib::ustring("\" \"") + fileName + Glib::ustring("\"");
+    const auto cmdLine = Glib::ustring("\"") + options.customEditorProg + "\" " + Glib::shell_quote(fileName);
     return spawnCommandAsync (cmdLine);
 
 #endif
